File: backend/app/schemas/paciente.py
from datetime import date
from typing import Literal

from pydantic import BaseModel


class PacienteCreate(BaseModel):
    nome_completo: str
    cpf: str
    telefone: str | None = None
    email: str | None = None
    data_nascimento: date | None = None
    convenio: str | None = None
    endereco: str | None = None


class PacienteUpdate(BaseModel):
    nome_completo: str | None = None
    telefone: str | None = None
    email: str | None = None
    data_nascimento: date | None = None
    convenio: str | None = None
    endereco: str | None = None
    status: Literal["ativo", "inativo"] | None = None


class PacienteResponse(BaseModel):
    id: int
    pessoa_id: int
    nome_completo: str
    cpf: str
    telefone: str | None
    email: str | None
    status: str
    data_nascimento: date | None
    convenio: str | None
    endereco: str | None

    model_config = {"from_attributes": True}

File: frontend/src/types/paciente.ts
export interface Paciente {
  id: number;
  pessoa_id: number;
  nome_completo: string;
  cpf: string;
  telefone: string | null;
  email: string | null;
  status: 'ativo' | 'inativo';
  data_nascimento: string | null;
  convenio: string | null;
  endereco: string | null;
}

export interface PacienteCreate {
  nome_completo: string;
  cpf: string;
  telefone?: string;
  email?: string;
  data_nascimento?: string;
  convenio?: string;
  endereco?: string;
}

export interface PacienteUpdate {
  nome_completo?: string;
  telefone?: string;
  email?: string;
  data_nascimento?: string;
  convenio?: string;
  endereco?: string;
  status?: 'ativo' | 'inativo';
}
